### PR TITLE
Answer:5 Solved crud challenge

### PR DESCRIPTION
**Pull request by pdarko99**

### CODE REVIEW

File: apps/crud/src/app/todo-item.component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+})
+export class TodoItemComponent {
+  @Input() todo!: Todo;
+  @Input() loading!: boolean;
+  @Output() deleteTodoEvent = new EventEmitter<number>();
+  @Output() updateTodoEvent = new EventEmitter<Todo>();
+
+  selectedId = 0;
+
+  status = '';
```
Comment (tomalaforge) [inline apps/crud/src/app/todo-item.component.ts:40]: this is not very reactive programming. A better solution will to create a local store to this component which will handle the loading and update and delete of ONE todo. This way, each line will have its own instance of a store, and you will not need to have this `selectedId` that is a big ugly, IMO.

**Comment by tomalaforge** [comment]
Thanks a lot for isolating the code of the challenge.

Can you rebase, I've moved the lib to create a better structure.

Great job but one of the underlying task was to play with DI to create local instance of a service to handle loading and update. this way you don't have to hack with selectedId.

**Comment by tomalaforge** [comment]
here is a small tuto to rebase easliy

Go inside your github repository and sync your fork repository
go inside your IDE
git checkout main
git pull 
git checkout [yourbranch]
git rebase main
correct conflict
git rebase --continue
git push -f

File: apps/crud/src/app/todo-item.component.ts (hunk 1 above)
Comment (pdarko99) [reply apps/crud/src/app/todo-item.component.ts]: okay I've made the changes.. please you can check it out and lemme have your thoughts

**Comment by pdarko99** [comment]
> here is a small tuto to rebase easliy
> 
> Go inside your github repository and sync your fork repository go inside your IDE git checkout main git pull git checkout [yourbranch] git rebase main correct conflict git rebase --continue git push -f

thanks very much

File: apps/crud/src/app/todo-item.component.ts (hunk 1 above)
Comment (tomalaforge) [reply apps/crud/src/app/todo-item.component.ts]: yes that's perfect, isn't it nicer ?

Comment (pdarko99) [reply apps/crud/src/app/todo-item.component.ts]: Yeah ...thanks very much

**Pull request closed without merging**